modules/post/windows/gather/wmic_command.rb

```diff
@@ -10,64 +10,81 @@ def initialize(info = {})
     super(
       update_info(
         info,
-        'Name' => 'Windows Gather Run Specified WMIC Command',
+        'Name' => 'Windows Gather Run WMIC Commands',
         'Description' => %q{
-          This module will execute a given WMIC command options or read
-          WMIC commands options from a resource file and execute the commands in the
-          specified Meterpreter session.
+          This module executes WMIC commands on the specified host.
         },
         'License' => MSF_LICENSE,
         'Author' => [ 'Carlos Perez <carlos_perez[at]darkoperator.com>'],
         'Platform' => [ 'win' ],
-        'SessionTypes' => [ 'meterpreter' ]
+        'SessionTypes' => [ 'meterpreter' ],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => []
+        }
       )
     )
 
-    register_options(
-      [
-        OptPath.new('RESOURCE', [false, 'Full path to resource file to read commands from.']),
-        OptString.new('COMMAND', [false, 'WMIC command options.']),
-      ]
-    )
+    register_options([
+      OptPath.new('RESOURCE', [false, 'Full path to resource file containing WMIC commands']),
+      OptString.new('COMMAND', [false, 'WMIC command']),
+    ])
   end
 
-  # Run Method for when run command is issued
   def run
-    tmpout = ''
-    if sysinfo.blank?
-      fail_with(Failure::BadConfig, "The session this module is running against doesn't support the sysinfo command!")
+    hostname = sysinfo.nil? ? cmd_exec('hostname') : sysinfo['Computer']
+    print_status("Running module against #{hostname} (#{session.session_host})")
+
+    resource_file = datastore['RESOURCE']
+    command = datastore['COMMAND']
+
+    if command.blank? && resource_file.blank?
+      fail_with(Failure::BadConfig, 'Please specify COMMAND or RESOURCE file.')
     end
-    print_status("Running module against #{sysinfo['Computer']}")
-    if datastore['RESOURCE']
-      if ::File.exist?(datastore['RESOURCE'])
 
-        ::File.open(datastore['RESOURCE']).each_line do |cmd|
-          next if cmd.strip.length < 1
-          next if cmd[0, 1] == '#'
+    commands = []
 
-          print_status "Running command #{cmd.chomp}"
+    if resource_file
+      fail_with(Failure::BadConfig, "Resource file #{resource_file} does not exist!") unless ::File.exist?(resource_file)
 
-          result = wmic_query(cmd.chomp)
-          store_wmic_loot(result, cmd)
-        end
-      else
-        raise 'Resource File does not exist!'
+      ::File.open(resource_file).each_line do |cmd|
+        next if cmd.strip.empty?
+        next if cmd.starts_with?('#')
+
+        commands << cmd.chomp
       end
+    else
+      commands << command
+    end
+
+    commands.uniq.each do |cmd|
```

Comment:
The `uniq` here might be unexpected for users

Just for visibility, what issue is this meant to mitigate? 🤔

Reply:
> The `uniq` here might be unexpected for users
> 
> Just for visibility, what issue is this meant to mitigate? thinking

No real issue. I'll remove it.